### DISCUSSION TITLE
M1 Mac Installation Troubleshooting Guide for Whisper, Ffmpeg and Brew

### DISCUSSION THREAD

**Discussion by brett-b112** [openai/whisper#1238] [General]
# Troubleshooting Installing Whisper

If `pip` or `pip3` is not found this could be a pathing issue try [article](https://www.freecodecamp.org/news/pip-command-not-found-mac-and-linux-error-solved/) article to solve this problem.
* sudo stands for: super user do 
   * This overrides any issues associated with permissions for downloading or modifying environment and pathing issues.

If you run into problems installing whisper with commands such as `pip install -U openai-whisper` or updating with `pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git` retry the commands using pip3

    pip3 install -U openai-whisper

Alternatively, the following command will pull and install the latest commit from this repository, along with its Python dependencies:

    pip3 install git+https://github.com/openai/whisper.git 

To update the package to the latest version of this repository, please run:

    pip3 install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git

If none of these work it is possible that this module was previously installed on your device, check with: 

pip list or pip3 list

search for whisper within the list to determine if it's already on your machine.

# Troubleshooting Installing Brew

Installing homebrew from the command line using the script provided from their official website [brew.sh](https://brew.sh/)

    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

After installing brew ensure that you read the last few lines and copied them into terminal to add to your current environment PATH. If you forgot to copy them modified versions are below.

    - Run these two commands in your terminal to add Homebrew to your PATH:
    #(echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/INSERT USER NAME HERE/.zprofile
    #eval "$(/opt/homebrew/bin/brew shellenv)"

Remove the `#` and insert your usr name in 'INSERT USER NAME HERE' before pasting in terminal to ensure that it is installed in the proper paths.

**If this did not solve the issue you can try this alternative method**

1. Open up users folder
2. Press `Command + Shift + .` to view the hidden folders
3. Open `.zshrc` folder 
4. Add `export PATH=/opt/homebrew/bin:$PATH` to the folder
5. Save and restart terminal
6. This should solve the issue and the environment can find brew now.

# Troubleshooting Installing Ffmpeg

Whispers README.md uses an outdated version of Ffmpeg with more functionality and modularity within the project. This can be found [here](https://github.com/kkroening/ffmpeg-python).

The full documentation of whisper can be found [here](https://ffmpeg.org/)

Not to many errors arise here if brew is installed correctly however, if you do refer to whisper README. 

If you encounter errors while pip installing:
* from [the rust website](https://www.rust-lang.org/learn/get-started) try: `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`

If rusts exists but is out of date try: `rustup update`

If you encounter further pathing errors try `export PATH="$HOME/.cargo/bin:$PATH"`

If `No module named 'setuptools_rust'` exists try:

`pip install setuptools-rust` or `pip3 install setuptools-rust`

This is a quick condensed trouble shooting guide for problems I faced and have seen online facing those trying to get Whisper installed on M1 Mac. 

*Thank you!*

**Comment by AlexDoehring** [comment]
Very Helpful Thanks!
